wireapp/wire-android
# fix: missing user status next to avatar [WPB-11191]

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-11191" title="WPB-11191" target="_blank"><img alt="Bug" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10803?size=medium" />WPB-11191</a>  [Android] Missing user status next to avatar
  </td></table>
  <br />
 



----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

The app doesn’t show user status dots next to avatars on conversation screen or conversation list.

### Causes (Optional)

There is a regression in current dev version, after implementing handling temporary users, the avatar composable was changed to accommodate that but the selection of `UserProfileAvatarType` has been changed so that these statuses are now not displayed.

### Solutions

I took the possibility to refactor `UserProfileAvatar` a bit, so that all indicators have its own composables and it's actually easier to add new ones to it. Whole avatar composable is now also easier to customise and calculations of paddings and/or border sizes are not that confusing - it just draws both and takes larger number to calculate overall size. Naming of `UserProfileAvatarType` items was probably also confusing, so it got renamed to better match the actual purposes. Status indicator border is now properly cut out from the avatar and its borders, so no need to draw any coloured borders that would need to always match the current background of the parent. `UserStatusIndicator` are also more adjusted for customisation and different sizes if needed.

### Testing

#### Test Coverage (Optional)


#### How to Test

Open the app on conversation list or open any specific conversation and check if user statuses are displayed properly.

----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
